dotnet/src/SemanticKernel.Core/Data/VolatileVectorStoreKeyResolver.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel.Data;
+
+/// <summary>
+/// Delegate that describes a function that given a record, finds the record key and returns it.
+/// </summary>
+/// <param name="record">The record to look up the key for.</param>
+/// <returns>The record key.</returns>
+[Experimental("SKEXP0001")]
+public delegate object? VolatileVectorStoreKeyResolver(object record);
```

Comment:
Should this delegate take a generic so we return the correct type? And possible a generic for the record?

Reply:
Thanks, updated.